src/Fmt_ast.ml

```diff
@@ -2810,47 +2810,25 @@ and fmt_class_type_field c ctx (cf : class_type_field) =
     $ fmt_atrs )
 
 and fmt_cases c ctx cs =
-  let rec pattern_len ?(parens = 0) pat =
-    match pat.ppat_desc with
-    | Ppat_any -> Some 1
-    | Ppat_var {txt= s; _}
-     |Ppat_constant (Pconst_integer (s, _))
-     |Ppat_constant (Pconst_float (s, _))
-     |Ppat_construct ({txt= Lident s; _}, None) ->
-        Some (String.length s)
-    | Ppat_construct ({txt= Lident s; _}, Some arg) -> (
-      match pattern_len ~parens:2 arg with
-      | Some arg -> Some (parens + String.length s + 1 + arg)
-      | None -> None )
-    | Ppat_constant (Pconst_char chr) ->
-        Some (String.length (char_escaped c ~loc:pat.ppat_loc chr) + 2)
-    | Ppat_variant (s, None) -> Some (String.length s + 1)
-    | Ppat_variant (s, Some arg) -> (
-      match pattern_len ~parens:2 arg with
-      | Some arg -> Some (parens + 1 + String.length s + 1 + arg)
-      | None -> None )
-    | Ppat_tuple ps ->
-        (* commas and parenthesis *)
-        let init = ((List.length ps - 1) * 2) + parens in
-        fold_pattern_len ~parens:2 ~init ~f:( + ) ps
-    | Ppat_alias _ | Ppat_interval _ | Ppat_construct _
-     |Ppat_constant (Pconst_string _)
-     |Ppat_record _ | Ppat_array _ | Ppat_constraint _ | Ppat_type _
-     |Ppat_or _ | Ppat_unpack _ | Ppat_lazy _ | Ppat_exception _
-     |Ppat_extension _ | Ppat_open _ ->
-        None
-  and fold_pattern_len ?parens ?(init = 0) ~f ps =
-    List.fold_until ~init ps
-      ~f:(fun acc pat ->
-        match pattern_len ?parens pat with
+  let pattern_len {pc_lhs; pc_guard; _} =
+    if Option.is_some pc_guard then None
+    else
+      let xpat = sub_pat ~ctx pc_lhs in
+      let fmted = Cmts.preserve (fmt_pattern c) xpat in
```

Comment:
We wanted to avoid doing that but I can't think of any valid reason. At least now it's more robust and handle every patterns.